tests/application-gateway-tests/test/executor/proxy/suite.go

```diff
@@ -0,0 +1,220 @@
+package proxy
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/kyma-project/kyma/tests/application-gateway-tests/test/executor/proxy/mock"
+
+	"github.com/kyma-project/kyma/tests/application-gateway-tests/test/executor"
+
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/kyma-project/kyma/tests/application-gateway-tests/test/tools"
+
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+)
+
+const (
+	defaultCheckInterval           = 2 * time.Second
+	appGatewayHealthCheckTimeout   = 15 * time.Second
+	accessServiceConnectionTimeout = 45 * time.Second
+	apiServerAccessTimeout         = 60 * time.Second
+	dnsWaitTime                    = 35 * time.Second
+
+	mockServiceNameFormat     = "%s-gateway-test-mock-service"
+	testExecutorPodNameFormat = "%s-tests-test-executor"
+)
+
+type TestSuite struct {
+	httpClient      *http.Client
+	k8sClient       *kubernetes.Clientset
+	podClient       corev1.PodInterface
+	serviceClient   corev1.ServiceInterface
+	config          executor.TestConfig
+	appMockServer   *mock.AppMockServer
+	mockServiceName string
+}
+
+func NewTestSuite(t *testing.T) *TestSuite {
+	config, err := executor.ReadConfig()
+	require.NoError(t, err)
+
+	k8sConfig, err := restclient.InClusterConfig()
+	require.NoError(t, err)
+
+	coreClientset, err := kubernetes.NewForConfig(k8sConfig)
+	require.NoError(t, err)
+
+	appMockServer := mock.NewAppMockServer(config.MockServicePort)
+
+	return &TestSuite{
+		httpClient:      &http.Client{},
+		k8sClient:       coreClientset,
+		podClient:       coreClientset.CoreV1().Pods(config.Namespace),
+		serviceClient:   coreClientset.CoreV1().Services(config.Namespace),
+		config:          config,
+		appMockServer:   appMockServer,
+		mockServiceName: fmt.Sprintf(mockServiceNameFormat, config.Application),
+	}
+}
+
+func (ts *TestSuite) Setup(t *testing.T) {
+	ts.WaitForAccessToAPIServer(t)
+
+	ts.appMockServer.Start()
+	ts.createMockService(t)
+
+	ts.CheckApplicationGatewayHealth(t)
+}
+
+func (ts *TestSuite) Cleanup(t *testing.T) {
+	t.Log("Calling cleanup")
+
+	err := ts.appMockServer.Kill()
+	assert.NoError(t, err)
+
+	ts.deleteMockService(t)
+}
+
+func (ts *TestSuite) ApplicationName() string {
+	return ts.config.Application
+}
+
+// WaitForAccessToAPIServer waits for access to API Server which might be delayed by initialization of Istio sidecar
```

Comment:
this comment seems a bit obsolete, we can remove it

Reply:
I do not think it is obsolete. I think it is quite useful as it seems to me that it is not totally obvious why we wait for access to API Server.

Reply:
Let's leave it then :)